File: apps/web/src/types/domains/resume.ts
/**
 * @file resume domain types
 * @description OpenAPI re-exports plus UI aliases for resume analysis cards.
 *
 * Analysis shapes use the generated contract as SSOT. The list/preview `Resume`
 * alias is ResumeResponse; the resume feature normalizes parsed_profile arrays
 * into ResumeItem before rendering.
 */

export type {
  ResumeAnalysis,
  DimensionScore,
  RewriteExample,
  SectionReview,
  RepoEvidence,
  RepoVerification,
  ResumePickerItem,
  ResumeResponse,
  CandidateProfile,
} from "@/lib/api/contract";

/** UI components keep the *Data suffix; underlying types are OpenAPI schemas. */
export type { ProjectCard as ProjectCardData } from "@/lib/api/contract";
export type { SkillTrust as SkillTrustData } from "@/lib/api/contract";
export type { CareerAnalysis as CareerAnalysisData } from "@/lib/api/contract";
export type { CompanyFit as CompanyFitData } from "@/lib/api/contract";

/** activate endpoint returns a full ResumeResponse. */
export type ResumeActivateResponse = import("@/lib/api/contract").ResumeResponse;

/** Wire type for HTTP payloads; feature UI uses ResumeItem after normalize. */
export type Resume = import("@/lib/api/contract").ResumeResponse;

File: apps/api/src/realmock/domains/resume/schemas/resume.py
"""Compatibility barrel for resume HTTP contracts.

Prefer ``realmock.domains.resume.schemas`` (package) or the split modules
(``analysis`` / ``request`` / ``response``). This module re-exports the same
public names so existing ``schemas.resume`` imports keep working.
"""

from realmock.domains.resume.schemas.analysis import (
    CareerAnalysis,
    CompanyFit,
    DimensionScore,
    ProjectCard,
    RepoEvidence,
    RepoVerification,
    ResumeAnalysis,
    RewriteExample,
    SectionReview,
    SkillTrust,
)
from realmock.domains.resume.schemas.request import ResumeAnalyzeRequest
from realmock.domains.resume.schemas.response import ResumeDomainLimits, ResumeResponse

__all__ = [
    "CareerAnalysis",
    "CompanyFit",
    "DimensionScore",
    "ProjectCard",
    "RepoEvidence",
    "RepoVerification",
    "ResumeAnalysis",
    "ResumeAnalyzeRequest",
    "ResumeDomainLimits",
    "ResumeResponse",
    "RewriteExample",
    "SectionReview",
    "SkillTrust",
]
